contracts/PartyBidFactory.sol

```diff
@@ -58,13 +61,15 @@ contract PartyBidFactory {
         // deploy logic contract
         PartyBid _logicContract = new PartyBid(_partyDAOMultisig, _tokenVaultFactory, _weth);
         // initialize logic contract
+        Structs.AddressAndAmount memory _split = Structs.AddressAndAmount(address(0), 0);
+        Structs.AddressAndAmount memory _tokenGate = Structs.AddressAndAmount(address(0), 0);
         _logicContract.initialize(
```

Comment:
I suppose this is to prevent someone calling `initialize()` later with a marketWrapper that selfdestructs, yeah? There's a cleaner/cheaper way to defend with something like:

```solidity
address immutable _IMPL;

constructor() public {
  _IMPL = address(this);
}

modifier onlyDelegateCall() {
  require(address(this) != _IMPL);
  _;
}

function bid() onlyDelegateCall ... { ... }
```

and stick

Reply:
this is a good idea 🤔

Reply:
though this would be cheaper than calling `initialize`, it doesn't seem like a cleaner option from a developer usability perspective.

it would be imperative that any functions which `delegateCall` use the `onlyDelegateCall` modifier, or else they would be vulnerable. whereas just initializing the contract at deploy time is a one-and-done solution that's imo harder to fuck up if modifying the PartyBid code in the future.

Reply:
x-posting from texts:
> you can use an immutable in PartyBid to remember the Factory and make sure only that contract can call `initialize`

imo this is a cleaner approach to ensuring that the logic contract doesn't self-destruct

Reply:
this was a great idea @merklejerk , it made our deploy script easier / cleaner (I don't have to find a dummy auction / token to intialize the logic contract anymore) and cut about 150k gas from the Factory deployment :~) 
https://github.com/PartyDAO/partybid/pull/61/commits/7452e798109d554731390fe787dc01fb41c7a376